client/config/crd/cbt.storage.k8s.io_snapshotmetadataservices.yaml

```diff
@@ -0,0 +1,67 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.15.0
+  name: snapshotmetadataservices.cbt.storage.k8s.io
+spec:
+  group: cbt.storage.k8s.io
+  names:
+    kind: SnapshotMetadataService
+    listKind: SnapshotMetadataServiceList
+    plural: snapshotmetadataservices
+    shortNames:
+    - sms
+    singular: snapshotmetadataservice
+  scope: Cluster
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: |-
+          SnapshotMetadataService is the Schema for the snapshotmetadataservices API
+          The presence of a SnapshotMetadataService CR advertises the existence of a CSI
+          driver's Kubernetes SnapshotMetadata gRPC service.
+          An audience scoped Kubernetes authentication bearer token must be passed in the
+          "security_token" field of each gRPC call made by a Kubernetes backup client.
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: |-
+              SnapshotMetadataServiceSpec defines the desired state of SnapshotMetadataService
+              This contains data needed to connect to a Kubernetes SnapshotMetadata gRPC service.
+            properties:
+              address:
+                description: The TCP endpoint address of the gRPC service.
+                type: string
```

Comment:
Shouldn't this be a list of addresses?

Reply:
No, as per the design we kept it as string 
https://github.com/kubernetes/enhancements/pull/4082/files#diff-1ac3e26395c19668afa33177d1d29013dc5b82ff86bad29ddd3249c969dabdf4R885

Reply:
I see that, and am wondering if we missed something there...

If you look at the [Service](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/) object the `clusterIPs` field is a slice.

If we leave it as a string and pass the Service object name, would the transport automatically handle multiple IPs?

Reply:
@carlbraganza yes, the service object name can be used as hostname to connect to sidecar (e.g [this](https://github.com/PrasadG193/external-snapshot-metadata/blob/9fd2496212e831ea53a471fef29d2156309e9b27/deploy/snapshotmetadataservice.yaml) is how we create it in prototype). Service object has 2 fields for IPs - `clusterIP` and `clusterIPs`. `clusterIPs` can have max 2 values (dual-stack) if multiple IP families are configured. Instead of IPs, I think service name as host makes sense since IP can be changed if service is recreated.

Reply:
Okay, we'll leave it as a string then, but perhaps we should change the description to say: "The DNS name of the snapshot metadata service" or some such thing?